### ISSUE TITLE
sync-users scripts fails with airflow 2.3.x and 2.4.x with 8.6.1 chart version

### ISSUE THREAD

**Issue by arundeep78** [airflow-helm/charts#703]
### Checks

- [X] I have checked for [existing issues](https://github.com/airflow-helm/charts/issues).
- [X] This report is about the [`User-Community Airflow Helm Chart`](https://github.com/airflow-helm/charts/tree/main/charts/airflow).

### Chart Version

8.6.1

### Kubernetes Version

```shell
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:58:16Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.5", GitCommit:"34f89fd3fb1a106e1b23d3454b2f2cbf305602a1", GitTreeState:"clean", BuildDate:"2023-02-10T14:48:59Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"linux/amd64"}
```

### Helm Version

```shell
version.BuildInfo{Version:"v3.8.0", GitCommit:"d14138609b01886f544b2025f5000351c9eb092e", GitTreeState:"clean", GoVersion:"go1.17.5"}
```

### Description

I upgraded airflow with below changes.
Component | Current Version | Target version
--- | --- | ---
Airflow helm chart | 8.5.3 | 8.6.1 (2.2.5-python3.8)
Airflow | 2.1.4-python3.8 | 2.4.3-python3.10

All other pods are working fine. Only 1 container `sync-users` keep on failing with " Multiple classes found for path "Permission" in the registry of this declarative base. Please use a fully module-qualified path".

other pods e.g. web, db_check and db_migrations also give the same warning message from configuration.py but they do not fail.

I have tried that with 3.9  or 3.8 python version as well and same error. I cannot directly go back airflow version as then db_migrations fails. 

I read other issues 607, 592  and 572.  I notice they are specific to 2.3.x but the issue I face seems to be exactly the same as reported by `softestplease` in [issue 607](https://github.com/airflow-helm/charts/issues/607#issuecomment-1252110091) 

Is there any work around for that or? 

### Relevant Logs

```shell
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  option = self._get_environment_variables(deprecated_key, deprecated_section, key, section)
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  option = self._get_environment_variables(deprecated_key, deprecated_section, key, section)
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:545 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
Traceback (most recent call last):
  File "/mnt/scripts/sync_users.py", line 142, in <module>
    flask_app = www_app.create_app()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/app.py", line 144, in create_app
    init_appbuilder(flask_app)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/extensions/init_appbuilder.py", line 648, in init_appbuilder
    AirflowAppBuilder(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/extensions/init_appbuilder.py", line 160, in __init__
    self.init_app(app, session)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/extensions/init_appbuilder.py", line 206, in init_app
    self.sm = self.security_manager_class(self)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/security.py", line 193, in __init__
    super().__init__(appbuilder)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/fab_security/sqla/manager.py", line 68, in __init__
    user_datamodel = SQLAInterface(self.user_model)
  File "/home/airflow/.local/lib/python3.10/site-packages/flask_appbuilder/models/sqla/interface.py", line 68, in __init__
    for prop in class_mapper(obj).iterate_properties:
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/base.py", line 442, in class_mapper
    mapper = _inspect_mapped_class(class_, configure=configure)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/base.py", line 421, in _inspect_mapped_class
    mapper._check_configure()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/mapper.py", line 1916, in _check_configure
    _configure_registries({self.registry}, cascade=True)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/mapper.py", line 3429, in _configure_registries
    _do_configure_registries(registries, cascade)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/mapper.py", line 3468, in _do_configure_registries
    mapper._post_configure_properties()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/mapper.py", line 1933, in _post_configure_properties
    prop.init()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/interfaces.py", line 231, in init
    self.do_init()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/relationships.py", line 2150, in do_init
    self._process_dependent_arguments()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/relationships.py", line 2245, in _process_dependent_arguments
    self.target = self.entity.persist_selectable
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py", line 1113, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/relationships.py", line 2112, in entity
    argument = self._clsregistry_resolve_name(self.argument)()
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/clsregistry.py", line 393, in _resolve_name
    rval = d[token]
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/util/_collections.py", line 746, in __missing__
    self[key] = val = self.creator(key)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/clsregistry.py", line 352, in _access_cls
    return _determine_container(key, decl_class_registry[key])
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/clsregistry.py", line 313, in _determine_container
    value = value.attempt_get([], key)
  File "/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/orm/clsregistry.py", line 144, in attempt_get
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "Permission" in the registry of this declarative base. Please use a fully module-qualified path.
Stream closed EOF for elyra-airflow/elyra-airflow-sync-users-5584bb9967-8wfxr (sync-airflow-users)
```

### Custom Helm Values

```yaml
airflow:
  # enable persistence of logs

  # Custom ingress to enable multiple hosts 
  hosts:
    - <hostname>

  # https://github.com/airflow-helm/charts/blob/main/charts/airflow/values.yaml

  ingress:
    enabled: false

  postgresql:
    enabled: false

  externalDatabase:
    host: airflow-db
    database: airflow
    user: airflow
    passwordSecret: elyra-airflow-db-password
    passwordSecretKey: password

  arcticdataDatabase:
    host: arcticdata-db
    database: arcticdata
    passwordSecret: arcticdata-db-password

  pgbouncer:
    enabled: false

  airflow:
    image:
      pullPolicy: IfNotPresent
      repository: apache/airflow
      tag: 2.4.3-python3.10
      gid: 0  
    executor: KubernetesExecutor
    config:
      AIRFLOW__KUBERNETES__WORKER_CONTAINER_REPOSITORY: "apache/airflow"
      AIRFLOW__KUBERNETES__WORKER_CONTAINER_TAG: "2.4.3-python3.10"
      AIRFLOW__KUBERNETES__LOGS_VOLUME_CLAIM: "airflow-logs"
      AIRFLOW__KUBERNETES__DAGS_VOLUME_CLAIM: "airflow"
      AIRFLOW__KUBERNETES__NAMESPACE: "airflow" # Deployment namespace
      AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "True"
      AIRFLOW__CORE__LOAD_EXAMPLES: "False"
      AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL: "300"
      AIRFLOW__KUBERNETES__GIT_SYNC_ONE_TIME: "False"
      AIRFLOW__LOGGING__REMOTE_LOGGING: "True"
      AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER: "s3://airflow-logs"
      AIRFLOW__LOGGING__REMOTE_LOG_CONN_ID: "minio"
    extraEnv:
      - name: MINIO_ACCESS_KEY
        valueFrom:
          secretKeyRef:
            name: minio-secret
            key: accessKey
      - name: MINIO_SECRET_KEY
        valueFrom:
          secretKeyRef:
            name: minio-secret
            key: secretKey
      
    connections:
      - id: minio
        type: s3
        description: Minio connection
        host: minio
        port: 9000
        login: ${MINIO_ACCESS_KEY}
        password: ${MINIO_SECRET_KEY}
        extra: |-
          { "aws_access_key_id": "${MINIO_ACCESS_KEY}",
            "aws_secret_access_key": "${MINIO_SECRET_KEY}",
            "host": "http://minio:9000",
            "region_name":"eu-central-1" }
    connectionsTemplates:
      MINIO_ACCESS_KEY:
        kind: secret
        name: minio-secret
        key: accessKey
      MINIO_SECRET_KEY:
        kind: secret
        name: minio-secret
        key: secretKey
    dbMigrations:
      enabled: true

  dags:
    path: /opt/airflow/dags
    gitSync:
      enabled: true #initial stop change it once verification is done
      repo: "git@github.com:<repo path>.git"
      branch: dev
      sshSecret: airflow-dags-ssh-secret
      sshSecretKey: id_rsa
      

  workers:
    enabled: false

  redis:
    enabled: false

  flower:
    enabled: false
```

**Comment by arundeep78** [comment]
I installed completely new 2.1.4-python3.8 with chart 8.6.1. I then upgraded to 2.3.2 -python3.10 and got the same error.

**Comment by arundeep78** [comment]
Continuing further I tried to have fresh install of  2.3.x on chart version 8.6.1. All versions 2.3.1 and above had the same issue. So, this does not seems to be related any db_migrations tasks.

 airflow 2.3.0 gives below error for all the pods because check_db container gives below error.
`raise AirflowConfigException(f"error: cannot use sqlite with the {self.get('core', '  executor')}")                                                                            airflow.exceptions.AirflowConfigException: error: cannot use sqlite with the KubernetesExecutor`

**Comment by thesuperzapper** [comment]
@arundeep78 my thoughts:

1. The values you provided look incorrectly indented, especially around `externalDatabase`, and `postgres`, I highly recommend starting from one of the sample files, for example the [`sample-values-CeleryExecutor.yaml`](https://github.com/airflow-helm/charts/blob/main/charts/airflow/sample-values-CeleryExecutor.yaml)
2. Because of 1, you were probably not installing into a clean database (as the embedded postgres PVC will have to be manually deleted from the namespace)
1. Can you try using a newer version of airflow 2.3, because 2.3.0 has many issues? (The latest is 2.3.4)

**Comment by arundeep78** [comment]
@thesuperzapper 
1. The values.yaml may look wrongly indented just in github. These worked fine on the old setup and it works fine for all the pods even on the new version. if you see any specific information in value,yaml that can break a very specific container's i.e. sync-users file sync-users.py file, then that would be helpful.
2. No, I am not using embedded postgres, this is configured by externalDatabase running postgres:15.2.
3. I mentioned above that I have tried upgrade to 2.3.x and also completely new installations on 2.3.X. Just that 2.3.0 fails with a different error as I mentioned above about sqllite. and all other fails with same error with sync-users,py file. I even tried 2.5.1 airflow and it has exactly same error.

I also referred to another issue where another user also reported the same issue.

The last airflow version that successfully works for completely new inatallation is 2.2.5-python3.9. I have not tried the upgrade yet.

But, till I get some another direction/hint/solution next steps are.
1. Try upgrade  to 2.2.5-python3.9
2. Try official airflow helm chart for 2.3.x and 2.5.1 and see what happens.

**Comment by arundeep78** [comment]
> 1. Try upgrade  to 2.2.5-python3.9

I install new 2.1.4-python3.8 and upgraded to 2.2.5-python3.9 and it works. 

I tried further upgrade to 2.3.4-python3.9 as sometimes 2 step upgrade works (although I was not positive in this case as clean install did not work) . As expected this upgrade failed with same container  failing with sync-users.py with same error message.

` sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "Permission" in the registry of this declarative base. Please use a fully module-qualified path. `

Which happens as mentioned in the main document in 
`File "/mnt/scripts/sync_users.py", line 142, in <module>
   flask_app = www_app.create_app()
File "/home/airflow/.local/lib/python3.9/site-packages/airflow/www/app.py", line 136,  in create_app                                                       init_appbuilder(flask_app)   `

**Comment by arundeep78** [comment]
I currently have another issue with "remote error: tls: bad record MAC" when using helm, so I could not continue the investigation. But, I will report back on new status soon. I may have an error, but cannot verify at the moment due to this tls issue.
